File: solution/2024. Maximize the Confusion of an Exam/2024.cpp
class Solution {
public:
    int maxConsecutiveAnswers(string answerKey, int k) {
        return max(count(answerKey, k, 'T'), count(answerKey, k, 'F'));
    }
    
    int count(const string& A, int k, char key) {
        int res = 0, i = 0;
        for (int j = 0; j < A.size(); ++j) {
            if (A[j] == key) {
                k--;
                while (k < 0) {
                    if (A[i] == key)
                        k++;
                    i++;
                }
            }
            res = max(res, j - i + 1);
        }
        return res;
    }
};

File: solution/2024. Maximize the Confusion of an Exam/2024.py
class Solution:
    def maxConsecutiveAnswers(self, A: str, k: int) -> int:
        def cnt(k: int, key: str) -> int:
            res, i = 0, 0
            for j in range(len(A)):
                if A[j] == key:
                    k -= 1
                    while k < 0:
                        if A[i] == key:
                            k += 1
                        i += 1
                res = max(res, j - i + 1)
            return res
        return max(cnt(k, 'T'), cnt(k, 'F'))
